pkg/virt-handler/node-labeller/node_labeller_test.go

```diff
@@ -111,64 +122,109 @@ var _ = Describe("Node-labeller ", func() {
-		testutils.ExpectNodePatch(kubeClient, v1.SEVESLabel)
 		res := nlController.execute()
 		Expect(res).To(BeTrue())
+
+		node := retriveNode(kubeClient)
+		Expect(node.Labels).To(HaveKey(v1.SEVESLabel))
 	})
 
 	It("should add usable cpu model labels for the host cpu model", func() {
-		testutils.ExpectNodePatch(kubeClient,
-			v1.HostModelCPULabel+"Skylake-Client-IBRS",
-			v1.CPUModelLabel+"Skylake-Client-IBRS",
-			v1.SupportedHostModelMigrationCPU+"Skylake-Client-IBRS",
-		)
 		res := nlController.execute()
 		Expect(res).To(BeTrue())
+
+		node := retriveNode(kubeClient)
+		Expect(node.Labels).To(SatisfyAll(
+			HaveKey(v1.HostModelCPULabel+"Skylake-Client-IBRS"),
+			HaveKey(v1.CPUModelLabel+"Skylake-Client-IBRS"),
+			HaveKey(v1.SupportedHostModelMigrationCPU+"Skylake-Client-IBRS"),
+		))
 	})
 
 	It("should add usable cpu model labels if all required features are supported", func() {
-		testutils.ExpectNodePatch(kubeClient,
-			v1.CPUModelLabel+"Penryn",
-			v1.SupportedHostModelMigrationCPU+"Penryn",
-		)
 		res := nlController.execute()
 		Expect(res).To(BeTrue())
+
+		node := retriveNode(kubeClient)
+		Expect(node.Labels).To(SatisfyAll(
+			HaveKey(v1.CPUModelLabel+"Penryn"),
+			HaveKey(v1.SupportedHostModelMigrationCPU+"Penryn"),
+		))
 	})
 
 	It("should not add usable cpu model labels if some features are not suported (svm)", func() {
-		testutils.DoNotExpectNodePatch(kubeClient,
-			v1.CPUModelLabel+"Opteron_G2",
-			v1.SupportedHostModelMigrationCPU+"Opteron_G2",
-		)
 		res := nlController.execute()
 		Expect(res).To(BeTrue())
+
+		node := retriveNode(kubeClient)
+		Expect(node.Labels).ToNot(SatisfyAny(
+			HaveKey(v1.CPUModelLabel+"Opteron_G2"),
+			HaveKey(v1.SupportedHostModelMigrationCPU+"Opteron_G2"),
+		))
+	})
+
+	It("should remove not found cpu model and migration model", func() {
+		node := retriveNode(kubeClient)
+		node.Labels[v1.CPUModelLabel+"Cascadelake-Server"] = "true"
+		node.Labels[v1.SupportedHostModelMigrationCPU+"Cascadelake-Server"] = "true"
+		node, err := kubeClient.CoreV1().Nodes().Update(context.TODO(), node, metav1.UpdateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		Expect(node.Labels).To(SatisfyAll(
+			HaveKey(v1.CPUModelLabel+"Cascadelake-Server"),
+			HaveKey(v1.SupportedHostModelMigrationCPU+"Cascadelake-Server"),
+		))
+
+		res := nlController.execute()
+		Expect(res).To(BeTrue())
+
+		node = retriveNode(kubeClient)
+		Expect(node.Labels).To(SatisfyAll(
+			HaveKey(v1.CPUModelLabel+"Skylake-Client-IBRS"),
+			HaveKey(v1.SupportedHostModelMigrationCPU+"Skylake-Client-IBRS"),
+		))
+		Expect(node.Labels).ToNot(SatisfyAny(
+			HaveKey(v1.CPUModelLabel+"Cascadelake-Server"),
+			HaveKey(v1.SupportedHostModelMigrationCPU+"Cascadelake-Server"),
+		))
 	})
 })
 
 func newNode(name string) *k8sv1.Node {
 	return &k8sv1.Node{
 		ObjectMeta: metav1.ObjectMeta{
 			Annotations: map[string]string{},
-			Labels:      map[string]string{},
+			Labels:      map[string]string{"INeedToBeHere": "trustme"},
```

Comment:
this label is added but not checked in any scenario. Please consider adding such a test.

Reply:
Good point. Added in the last commit.